element-connector/src/main/java/org/eclipse/californium/elements/RawData.java

```diff
@@ -327,7 +226,7 @@ public EndpointContext getEndpointContext() {
 	 *         otherwise
 	 */
 	public boolean isSecure() {
-		return (endpointContext != null && endpointContext.get(DtlsEndpointContext.KEY_SESSION_ID) != null);
+		return (peerEndpointContext != null && peerEndpointContext.get(DtlsEndpointContext.KEY_SESSION_ID) != null);
```

Comment:
I suspect this implementation of isSecure is not really ok or maybe this is just the name and/or javadoc which is not ok.

Now we are able to set Endpointcontext at constructor and as this context could be incomplete, there could be other condition we make this secure. (e.g. a principal is available)

I check and this is called only on CoapEndpoint.InboxImpl.receiveRequest(Request, RawData), so I don't think this is so impacting. (so probably not so important)

Reply:
There was a short discussion about the isSecure() in #156.

In the end, I'm not sure, if this is still relevant. This would only make sense, if a CoapEndpoint would receive mixed request. But currently we use separate connectors and endpoints for coap or coaps.

Reply:
I continue the discussion in #156, because it's more appropriate.

Reply:
I would postpone that discussion, because I think "isSecure()" will be better removed, and the scheme is better set using the scheme from the connector. But to check, if I have overseen a use-case, I would first try to finish the changes for the endpoint context.
The next step would be to switch to that EndpointContext in the core.